pkg/proc/core/linux_core.go

```diff
@@ -29,15 +29,69 @@ const NT_X86_XSTATE elf.NType = 0x202 // Note type for notes containing X86 XSAV
 // NT_AUXV is the note type for notes containing a copy of the Auxv array
 const NT_AUXV elf.NType = 0x6
 
+// NT_FPREGSET is the note type for floating point registers.
+const NT_FPREGSET elf.NType = 0x2
+
+// Fetch architecture using exeELF.Machine from core file
+// Refer http://man7.org/linux/man-pages/man5/elf.5.html
+const EM_AARCH64 = 183
+const EM_X86_64 = 62
+
+const (
+	_ARM_FP_HEADER_START = 512
+)
+
 const elfErrorBadMagicNumber = "bad magic number"
 
-// readLinuxAMD64Core reads a core file from corePath corresponding to the executable at
+func (lastThread *linuxAMD64Thread) linuxThreadsFromNotesAMD64(p *Process, notes []*Note) (*Process, error) {
```

Comment:
No reason to have lastThread as a receiver here, it can just be a local variable. Also, no reason to return a *Process since p is already a pointer.